notes-service/src/main/java/org/exoplatform/wiki/jpa/search/WikiElasticSearchServiceConnector.java

```diff
@@ -111,33 +120,67 @@ protected String getSourceFields() {
     return StringUtils.join(sourceFields, ",");
   }
 
-  public List<SearchResult> searchWiki(String searchedText, String userId, boolean isFavorites, int offset, int limit) {
-    List<SearchResult> searchResults = filteredWikiSearch(searchedText, userId, isFavorites, offset, limit);
-    return searchResults;
+  public List<SearchResult> searchWiki(String searchedText, String userId, List<String> tagNames, boolean isFavorites, int offset, int limit) {
+      return filteredWikiSearch(searchedText, userId, tagNames, isFavorites, offset, limit);
   }
 
-  protected List<SearchResult> filteredWikiSearch(String query, String userId, boolean isFavorites, int offset, int limit) {
+  protected List<SearchResult> filteredWikiSearch(String query, String userId, List<String> tagNames, boolean isFavorites, int offset, int limit) {
     Set<String> ids = getUserSpaceIds(userId);
-    String esQuery = buildQueryStatement(ids, userId, query, isFavorites, offset, limit);
+    String esQuery = buildQueryStatement(ids, userId, tagNames, query, isFavorites, offset, limit);
     String jsonResponse = getClient().sendRequest(esQuery, getIndex());
     return buildWikiResult(jsonResponse);
   }
 
+  private String buildTagsQueryStatement(List<String> values) {
+    if (values == null || values.isEmpty()) {
+      return "";
+    }
+    List<String> tagsQueryParts = values.stream()
+            .map(value -> "{\"term\": {\n" +
```

Comment:
Same, you can use `triple double quote sequence` to simplify reading and maintenance

Reply:
done